### PR TITLE
tidb-configuration-file: add graceful-wait-before-shutdown

### PR DESCRIPTION
**Pull request by dveeden**
Closes: #5823

### What is changed, added or deleted? (Required)

tidb-configuration-file: add graceful-wait-before-shutdown

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.1 (TiDB 5.1 versions)
- [x] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### CODE REVIEW

**Review by morgo — APPROVED** [review]

**Review by bb7133 — APPROVED** [review]

**Comment by bb7133** [comment]
I'm not a reviewer, PTAL @TomShawn

File: tidb-configuration-file.md [hunk 1]
```diff
@@ -171,6 +171,11 @@ The TiDB configuration file supports more options than command-line parameters.
 + Default value: `true`
 + When this configuration value is `true`, the maximum length of a single `ENUM` element and a single `SET` element is 255 characters, which is compatible with [MySQL 8.0](https://dev.mysql.com/doc/refman/8.0/en/string-type-syntax.html). When this configuration value is `false`, there is no limit on the length of a single element, which is compatible with TiDB (earlier than v5.0).
 
+#### `graceful-wait-before-shutdown` <span class="version-mark">New in v5.0</span>
+
+- Wait for n seconds when shutting down to allow clients to disconnect.
```
Comment (TomShawn) [inline tidb-configuration-file.md:176]: any default value?

Comment (dveeden) [reply tidb-configuration-file.md:176]: Yes. Added that.

Comment (TomShawn) [reply tidb-configuration-file.md:176]:
```suggestion
- Specifies the number of seconds that TiDB waits when you shut down the server, which allows the clients to disconnect.
```

Comment (dveeden) [reply tidb-configuration-file.md:176]: @TomShawn did you mean to resolve this conversation? or do you still want me to apply the suggestion?

File: tidb-configuration-file.md [hunk 2]
```diff
@@ -171,6 +171,12 @@ The TiDB configuration file supports more options than command-line parameters.
...
+- Default value: `0`
+- While in the grace period the HTTP status will already indicate failure to allow loadbalancers to reroute traffic.
```
Comment (TomShawn) [inline tidb-configuration-file.md:178]:
```suggestion
- When TiDB is waiting for shutdown (in the grace period), the HTTP status will indicate a failure, which allows the load balancers to reroute traffic.
```

File: tidb-configuration-file.md (hunk 1 above)
Comment (TomShawn) [reply tidb-configuration-file.md:176]: I find myself unable to comment on this line unless I do it here😂 Please see whether my comment can be accepted or not. Thanks!

File: tidb-configuration-file.md (hunk 2 above)
Comment (dveeden) [reply tidb-configuration-file.md]: Yes correct.

**Review by TomShawn — APPROVED** [review]
LGTM

**Comment by TomShawn** [comment]
> I'm not a reviewer, PTAL @TomShawn

In fact you are a reviewer no matter whether the ti-chi-bot counts you or not😂.

**Pull request merged**